lib/cookie/secureContext.ts

```diff
@@ -0,0 +1,140 @@
+import { URL } from 'url';
+import { isIP, isIPv4, isIPv6 } from 'net';
+
+/**
+ * Checks if the given IPv4 address is in the 127.0.0.0/8 loopback range.
+ *
+ * @remarks
+ * @param address - The IPv4 address to check.
+ * @returns `true` if the address is loopback, otherwise `false`.
+ */
+function isLoopbackV4(address: string): boolean {
+  // 127.0.0.0/8: first octet = 127
+  const octets = address.split('.');
+  return (
+    octets.length === 4 &&
+    octets[0] !== undefined &&
+    parseInt(octets[0], 10) === 127
+  );
+}
+
+/**
+ * Checks if the given IPv6 address is the loopback address (`::1`).
+ *
+ * @remarks
+ * @param address - The IPv6 address to check.
+ * @returns `true` if the address is loopback, otherwise `false`.
+ */
+function isLoopbackV6(address: string): boolean {
+  // new URL(...) follows the WHATWG URL Standard
+  // which compresses IPv6 addresses, therefore the IPv6
+  // loopback address will always be compressed to '[::1]':
+  // https://url.spec.whatwg.org/#concept-ipv6-serializer
+  return (address === '::1');
+}
+
+/**
+ * Determines if the given address (IPv4 or IPv6) is a loopback address.
+ *
+ * @remarks
+ * @param address - The IP address to check.
+ * @returns `true` if the address is loopback, otherwise `false`.
+ */
+function isIpLoopback(address: string): boolean {
+  if (isIPv4(address)) {
+    return isLoopbackV4(address);
+  }
+
+  if (isIPv6(address)) {
+    return isLoopbackV6(address);
+  }
+
+  return false;
+}
+
+/**
+ * Checks if the given lowercase host ends with the `.localhost` TLD.
+ *
+ * @remarks
+ * @param lowerHost - The lowercase host string to check.
+ * @returns `true` if the host ends with `.localhost`, otherwise `false`.
+ */
+function isNormalizedLocalhostTLD(lowerHost: string): boolean {
+  return lowerHost.endsWith('.localhost');
+}
+
+/**
+ * Checks if the given host is `localhost` or matches `*.localhost`.
+ *
+ * @remarks
+ * @param host - The host string to check.
+ * @returns `true` if the host is considered local, otherwise `false`.
+ */
+function isLocalHostname(host: string): boolean {
+  const lowerHost = host.toLowerCase();
+  return lowerHost === 'localhost' ||
+    isNormalizedLocalhostTLD(lowerHost);
+}
+
+/**
+ * Removes leading and trailing square brackets if present.
+ * Adapted from https://github.com/chromium/chromium/blob/main/url/gurl.cc#L440-L448
+ *
+ * @param host - The host string to process.
+ * @returns The host string without leading and trailing square brackets if previously present.
+ */
+function hostNoBrackets(host: string): string {
+  if (host.length >= 2 && host.startsWith('[') && host.endsWith(']')) {
```

Comment:
```suggestion
  // Adapted from https://github.com/chromium/chromium/blob/main/url/gurl.cc#L440-L448
  if (host.length >= 2 && host.startsWith('[') && host.endsWith(']')) {
```

Inline where you sourced your adaptation.